### PR TITLE
Add logger to the CommandHistory class

### PR DESCRIPTION
**Pull request by weijianwong**
Add logging to the CommandHistory class to track when users click on the up or down arrow keys when interacting with the command box.

### CODE REVIEW

File: src/main/java/tuteez/logic/commands/CommandHistory.java [hunk 1]
```diff
@@ -18,6 +24,7 @@ public class CommandHistory {
     public CommandHistory() {
         this.history = new ArrayList<>();
         this.currentIndex = -1;
+        logger.info("CommandHistory initialized with empty history.");
```
Comment (potaotototo) [inline src/main/java/tuteez/logic/commands/CommandHistory.java:27]: Good use of logger!

File: src/main/java/tuteez/logic/commands/CommandHistory.java [hunk 2]
```diff
@@ -37,8 +45,11 @@ public String getPreviousCommand() {
         assert currentIndex >= -1 && currentIndex <= history.size();
         if (currentIndex > 0) {
             currentIndex--;
-            return history.get(currentIndex);
+            String previousCommand = history.get(currentIndex);
+            logger.info("Retrieved previous command: " + previousCommand);
+            return previousCommand;
         }
+        logger.warning("No previous command available.");
```
Comment (potaotototo) [inline src/main/java/tuteez/logic/commands/CommandHistory.java:52]: Good use of the Logger class, shows expertise and knowledge. 

Good job!

File: src/main/java/tuteez/logic/commands/CommandHistory.java [hunk 3]
```diff
@@ -49,9 +60,12 @@ public String getNextCommand() {
         assert currentIndex >= -1 && currentIndex <= history.size();
         if (currentIndex < history.size() - 1) {
             currentIndex++;
-            return history.get(currentIndex);
+            String nextCommand = history.get(currentIndex);
+            logger.info("Retrieved next command: " + nextCommand);
+            return nextCommand;
         }
         currentIndex = history.size();
+        logger.warning("No next command available. Resetting to end of history.");
```
Comment (potaotototo) [inline src/main/java/tuteez/logic/commands/CommandHistory.java:68]: Resetting to end of history sounds a bit vague, perhaps try rephrasing?

File: src/main/java/tuteez/logic/commands/CommandHistory.java (hunk 2 above)
Comment (luileng) [inline src/main/java/tuteez/logic/commands/CommandHistory.java:52]: Perhaps you could use `logger.info` here instead?

**Pull request merged**